doc/source/commands/communication.rst

```diff
@@ -0,0 +1,129 @@
+
+Accessing processors
+~~~~~~~~~~~~~~~~~~~~
+
+The easiest way of accessing the processor's :struct:`kOSProcessor` structure (as long as your CPUs have their
+:ref:`name tags <nametag>` set) is to use the following function:
+
+.. function:: PROCESSOR(volumeOrNameTag)
+
+    :parameter volumeOrNameTag: (:struct:`Volume` | `String`) can be either an instance of :struct:`Volume` or a string
+
+    Depending on the type of the parameter value will either return the processor associated with the given
+    :struct:`Volume` or the processor with the given name tag.
+
+A list of all processors can be obtained using the :ref:`List <list>` command::
+
+  LIST PROCESSORS IN ALL_PROCESSORS.
+  PRINT ALL_PROCESSORS[0]:NAME.
+
+Finally, processors can be accessed directly, like other :ref:`parts and modules <part>`::
+
+  PRINT SHIP:MODULESNAMED("kOSProcessor")[0]:VOLUME:NAME.
+
+Sending messages
+~~~~~~~~~~~~~~~~
+
+Once we have a :struct:`kOSProcessor` structure associated with the processor we want to send the message to we can
+easily obtain a :struct:`Connection` to that processor using :attr:`kOSProcessor:CONNECTION`. Next we're going to
+send a message using :meth:`Connection:SENDMESSAGE`. This is an example of how the whole thing could look::
+
+  SET MESSAGE TO "undock". // can be any serializable value or a primitive
+  SET P TO PROCESSOR("probe").
+  IF P:SENDMESSAGE(MESSAGE) {
+    PRINT "Message sent!".
+  }
+
+Receiving messages
+~~~~~~~~~~~~~~~~~~
+
+We now switch to the second CPU (in the example above it had a name tag `"probe."`). It should have a message
+in its message queue. To access the queue from the current processor we use the :attr:`CORE:MESSAGES` suffix.
+The following code will initiate undocking after the `"undock"` message is received::
+
+  WAIT UNTIL NOT CORE:MESSAGES:EMPTY. // make sure we've received something
+  SET RECEIVED TO CORE:MESSAGES:POP.
+  IF RECEIVED:CONTENT = "undock" {
+    PRINT "Undocking!!!".
+    UNDOCK().
+  } ELSE {
+    PRINT "Unexpected message: " + RECEIVED:CONTENT.
+  }
+
+
+Inter-vessel communication
+--------------------------
+
+This will be very similar to how inter-processor communication was done. We'll just have to use
+:struct:`vessel(vesselName) <Vessel>` to find the vessel we want to send the message to
+and then :attr:`Vessel:CONNECTION` to get the connection. This is how sender's code could look like::
+
+
+  SET MESSAGE TO "HELLO". // can be any serializable value or a primitive
+  SET C TO VESSEL("SOMEVESSEL"):CONNECTION.
+  PRINT "Delay is " + C:DELAY + " seconds".
+  IF C:SENDMESSAGE(MESSAGE) {
+    PRINT "Message sent!".
+  }
+
+The CPU on the receiving vessel will use :attr:`SHIP:MESSAGES <Vessel:MESSAGES>` to access the vessel's message queue::
+
+  WHEN NOT SHIP:MESSAGES:EMPTY {
+    SET RECEIVED TO SHIP:MESSAGES:POP. // or do something else...
+    PRINT "Sent by " + RECEIVED:SENDER:NAME + " at " + RECEIVED:SENTAT.
+    PRINT RECEIVED:CONTENT.
+  }
```

Comment:
I suspect more people will be interested in inter-vessel than intra-vessel comms.  (I know I am).  It might make more sense to reverse the order of explaining things.  (Give inter-vessel comms first, then explain how intra-vessel comms differ.)  Also, this gets the user to see useful code faster, higher on the page which is usually a good thing.  Intra-vessel comms have the extra complexity of explaining how to get which processor you're talking about, while inter-vessel comms only need people to know how to get a vessel by name, and know that ``SHIP`` is the name of the current vessel.  (So not only do I think it's the more useful one of the two, I also think it takes less text to explain it which would be a second reason to put it first).